azurelinuxagent/ga/exthandlers.py

```diff
@@ -329,36 +322,40 @@ def format_value(parse_fn, value):
         return activity_id, correlation_id, gs_creation_time
 
     def run(self):
-
         try:
             self.ext_handlers, etag = self.protocol.get_ext_handlers()
-            msg = u"Handle extensions updates for incarnation {0}".format(etag)
-            logger.verbose(msg)
-            # Log status report success on new config
-            self.log_report = True
-            incarnation_changed = self._incarnation_changed(etag)
-            if self._extension_processing_allowed() and incarnation_changed:
-                activity_id, correlation_id, gs_creation_time = self.get_goal_state_debug_metadata()
-
-                logger.info(
-                    "ProcessGoalState started [Incarnation: {0}; Activity Id: {1}; Correlation Id: {2}; GS Creation Time: {3}]".format(
-                        etag, activity_id, correlation_id, gs_creation_time))
 
-                self.__process_and_handle_extensions(etag)
-                self.last_etag = etag
+            if not (self._extension_processing_allowed()):
+                return
 
-            self.report_ext_handlers_status(incarnation_changed=incarnation_changed)
```

Comment:
I separated the calls to run and report_ext_handlers_status.

We need to report status even is there is no extension processing. Reporting status should be separate from extension handling altogether but currently there are several dependencies, mainly in the tests (even this simple refactoring affected 100+ tests). Since we are abandoning this code base I did not refactor further.